src/components/widgets/Toggle.tsx

```diff
@@ -0,0 +1,235 @@
+import { CSSProperties } from "react";
+import { Subset } from "../../definitions/types";
+
+import { ThemeRole } from "../theming/types";
+import { useMemo, useState } from "react";
+
+// Definitions
+import { primaryFont } from "../../styleDefinitions/typography";
+import { blue } from "@material-ui/core/colors";
+import { gray } from "../../definitions/colors";
+import { useUITheme } from "../theming";
+import { merge } from "lodash";
+
+// Type definitions for working with styles.
+type ToggleColorStyleSpec = {
+  backgroundColor: CSSProperties["color"];
+  labelColor: CSSProperties["color"];
+  knobColor: CSSProperties["color"];
+  borderColor: CSSProperties["color"];
+};
+
+type ToggleStateStyleSpec = {
+  off: ToggleColorStyleSpec;
+  on: ToggleColorStyleSpec;
+};
+
+type ToggleStyleSpec = {
+  container: React.CSSProperties;
+  default: ToggleStateStyleSpec;
+  hover: ToggleStateStyleSpec;
+  disabled: ToggleStateStyleSpec;
+};
+
+type ToggleStyleSpecSubset = Subset<ToggleStyleSpec>;
+
+// Prop definitions.
+export type ToggleProps = {
+  /** Whether the toggle is off or on. */
+  state: boolean;
+  /** Callback to invoke when the toggle is flipped. */
+  onToggle: (state: boolean) => void;
+  /** Whether the component is currently disabled for user interactions. Optional. */
+  disabled?: boolean;
+  /** Text to render beside the toggle. Optional. */
+  label?: string;
```

Comment:
We can change this type to `ReactNode` so that we can accommodate any styling or tooltips a label may have.

This may require us to add an `ariaLabel` prop typed as `string` since pulling the text content out of a react node may add unnecessary complexity.